Tests/GravityTest.php

```diff
@@ -0,0 +1,114 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Orbitale\Component\ImageMagick\Tests;
+
+use Orbitale\Component\ImageMagick\Command;
+use Orbitale\Component\ImageMagick\ReferenceClasses\Gravity;
+
+class GravityTest extends AbstractTestCase
+{
+    /**
+     * @param string $gravity
+     *
+     * @dataProvider provideValidGravities
+     */
+    public function testGravity($gravity): void
+    {
+        $gravity = new Gravity($gravity);
+
+        try {
+            $validatedGravity = $gravity->validate();
+        } catch (\Exception $e) {
+            static::fail($e->getMessage());
+
+            return;
+        }
+
+        static::assertIsString($validatedGravity);
+        static::assertNotEmpty($validatedGravity);
+
+        if ('' === $validatedGravity) {
+            static::markTestSkipped('No gravity to check. ['.$validatedGravity.'] ['.\implode(',', \func_get_args()).']');
+        }
+
+        $command = new Command(IMAGEMAGICK_DIR);
+
+        $outputFile = $this->resourcesDir.'/outputs/moon_180_test_gravity_'.\md5($gravity.'test_geo').'.jpg';
+
+        if (\file_exists($outputFile)) {
+            \unlink($outputFile);
+        }
+
+        $command
+            ->convert($this->resourcesDir.'/moon_180.jpg')
+            ->gravity($gravity)
+            ->resize('100x100')
+            ->file($outputFile, false)
+        ;
+
+        $response = $command->run();
+
+        static::assertTrue($response->isSuccessful(), \sprintf(
+            "Gravity fixture \"%s\" returned an ImageMagick error.\nFull command: %s\nErrors:\n%s\n%s",
+            $gravity->validate(),
+            $command->getCommand(),
+            $response->getOutput(),
+            $response->getError()
+        ));
+        static::assertFileExists($outputFile);
+    }
+
+    public function provideValidGravities(): ?\Generator
+    {
+        yield 0 => ["NorthWest"];
+        yield 1 => ["North"];
+        yield 2 => ["NorthEast"];
+        yield 3 => ["West"];
+        yield 4 => ["Center"];
+        yield 5 => ["East"];
+        yield 6 => ["SouthWest"];
+        yield 7 => ["South"];
+        yield 8 => ["SouthEast"];
+    }
+
+    /**
+     * @param string $gravity
+     *
+     * @dataProvider provideWrongGravities
+     */
+    public function testWrongGravities($gravity): void
+    {
+        $testGravity = new Gravity($gravity);
+
+        $message = null;
+
+        try {
+            $testGravity->validate();
```

Comment:
You can use `$this->expectException(\InvalidArgumentException::class);` and `$this->expectExceptionMessage('...');` instead

Reply:
Resolved in commit 1a10dd2.